### PR TITLE
feat: add seam cli code samples

### LINKED ISSUE #49: Add Seam CLI code samples
The key should be `seam_cli`, but we need to add a new `language` property since the cli runs in bash. Doc gen should be updated to use the new property

```
const CodeSampleSchema = CodeSampleDefinitionSchema.extend({
  code: z.record(
    z.enum(['javascript', 'python', 'ruby']),
    z.object({
      title: z.string().min(1),
      language: z.enum(['javascript', 'python', 'ruby', 'bash']),
      request: z.string(),
      response: z.string(),
    }),
  ),
})

### PR DESCRIPTION
**Pull request by mikewuu**
closes #49 

Here's what this generates.

CleanShot 2024-08-13 at 00 54 04@2x

Is there anything else required for #49?

### CODE REVIEW

File: src/lib/code-sample/schema.ts [hunk 1]
```diff
@@ -79,6 +80,11 @@ export const createCodeSample = (
         request: createPhpRequest(codeSampleDefinition, context),
         response: createPhpResponse(codeSampleDefinition, context),
       },
+      bash: {
+        title: 'Bash',
```
Comment (razor-x) [inline src/lib/code-sample/schema.ts:84]:
```suggestion
        title: 'Seam CLI',
```

Comment (razor-x) [inline src/lib/code-sample/schema.ts:84]:
```suggestion
        request_syntax: 'bash',
        response_syntax: 'json',
```

File: src/lib/code-sample/schema.ts [hunk 2]
```diff
@@ -37,7 +38,7 @@ export type CodeSampleDefinition = z.output<typeof CodeSampleDefinitionSchema>
 
 const CodeSampleSchema = CodeSampleDefinitionSchema.extend({
   code: z.record(
-    z.enum(['javascript', 'python', 'php', 'ruby']),
+    z.enum(['javascript', 'python', 'php', 'ruby', 'bash']),
     z.object({
       title: z.string().min(1),
```
Comment (razor-x) [inline src/lib/code-sample/schema.ts:43]: Let's add a new enum to pass though to the markdown code block,

```ts
const syntax = z.enum(['javascript', 'python', 'php', 'ruby', 'bash'])
```

```suggestion
      title: z.string().min(1),
      request_syntax: syntax,
      response_syntax: syntax,
```

File: src/lib/code-sample/bash.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+): string => {
+  const parts = request.path.split('/')
+  const requestParams = Object.entries(request.parameters)
+    .map(([key, value]) => `--${key} "${JSON.stringify(value)}"`)
+    .join(' ')
+
+  return `seam${parts.map((p) => kebabCase(p)).join(' ')} ${requestParams}`
+}
+
+export const createBashResponse = createJsonResponse
```
Comment (razor-x) [inline src/lib/code-sample/bash.ts:18]:
```suggestion
export const createSeamCliResponse = createJsonResponse
```

File: src/lib/code-sample/schema.ts (hunk 1 above)
Comment (mikewuu) [reply src/lib/code-sample/schema.ts]: updated in https://github.com/seamapi/blueprint/pull/58/commits/0b1f37a41acad1ce4a296178b333d1b07f8c20b7

File: src/lib/code-sample/schema.ts (hunk 2 above)
Comment (mikewuu) [reply src/lib/code-sample/schema.ts:48]: updated in https://github.com/seamapi/blueprint/pull/58/commits/0b1f37a41acad1ce4a296178b333d1b07f8c20b7

File: src/lib/code-sample/schema.ts [hunk 4]
```diff
@@ -63,21 +71,36 @@ export const createCodeSample = (
         title: 'JavaScript',
         request: createJavascriptRequest(codeSampleDefinition, context),
         response: createJavascriptResponse(codeSampleDefinition, context),
+        request_syntax: 'javascript',
+        response_syntax: 'json',
```
Comment (razor-x) [inline src/lib/code-sample/schema.ts:75]:
```suggestion
        response_syntax: 'javascript',
```

File: src/lib/code-sample/schema.ts [hunk 5]
```diff
@@ -63,21 +71,36 @@ export const createCodeSample = (
...
         response: createRubyResponse(codeSampleDefinition, context),
+        request_syntax: 'ruby',
+        response_syntax: 'ruby',
       },
       php: {
         title: 'PHP',
         request: createPhpRequest(codeSampleDefinition, context),
         response: createPhpResponse(codeSampleDefinition, context),
+        request_syntax: 'php',
+        response_syntax: 'json',
```
Comment (razor-x) [inline src/lib/code-sample/schema.ts:96]: Not for this PR, but this really should be php

File: test/snapshots/blueprint.test.ts.md [hunk 6]
```diff
@@ -70,27 +70,42 @@ Generated by [AVA](https://avajs.dev).
...
+                      request_syntax: 'ruby',
                       response: `<Seam::Foo:0x00000␊
                       foo_id="8d7e0b3a-b889-49a7-9164-4b71a0506a33"␊
                       name="Best foo"␊
                       nullable_property=nil>`,
+                      response_syntax: 'ruby',
                       title: 'Ruby',
                     },
+                    seam_cli: {
+                      request: 'seam foos get --foo_id ""8d7e0b3a-b889-49a7-9164-4b71a0506a33""',
```
Comment (razor-x) [inline test/snapshots/blueprint.test.ts.md:103]: Not sure what's going on with the double `""`.

**Review by razor-x — APPROVED** [review]

File: src/lib/code-sample/schema.ts (hunk 4 above)
Comment (mikewuu) [reply src/lib/code-sample/schema.ts:75]: Is it javascript, though? this one's a hard one 😖

File: src/lib/code-sample/schema.ts (hunk 5 above)
Comment (mikewuu) [reply src/lib/code-sample/schema.ts:96]: https://github.com/seamapi/blueprint/issues/60

File: test/snapshots/blueprint.test.ts.md (hunk 6 above)
Comment (mikewuu) [reply test/snapshots/blueprint.test.ts.md]: Fixed in https://github.com/seamapi/blueprint/pull/58/commits/4ae96fd18940c33f039d1a7d5c0f53cd989f3c02

**Review by razor-x — APPROVED** [review]

**Pull request merged**